tensorflow_addons/optimizers/weight_decay_optimizers.py

```diff
@@ -170,7 +175,8 @@ def _resource_apply_sparse(self, grad, var, indices):
             return super()._resource_apply_sparse(grad, var, indices)
 
 
-def extend_with_decoupled_weight_decay(base_optimizer):
+def extend_with_decoupled_weight_decay(
+        base_optimizer: tf.keras.optimizers) -> tf.keras.optimizers:
```

Comment:
tf.keras.optimizer is a module, not a class. I think you mean `tf.keras.optimizers.Optimizer`. Also you can use the typechecked decorator here since there is no risk for this function to be converted using autograph.

Reply:
Is this another case to Union string and optimizer since Keras accepts either or?

Reply:
Nope, not here. There is a subclassing done just after.

Reply:
Yes, thanks @gabrieldemarmiesse. I've tried, but have the following error msg, `TypeError: type of argument "base_optimizer" must be tensorflow.python.keras.optimizer_v2.optimizer_v2.OptimizerV2; got tensorflow.python.keras.optimizer_v2.gradient_descent.SGD instead`. I'm not sure why can't pass the test.

Reply:
I get it now! It's a class, not an optimizer.

That's a strange API but ok. I don't even know how to put typing hints. I'm looking for something like `typing.class` but I can't find it.

Reply:
I am searching for too, but no idea yet, should I use `tf.keras.optimizers` instead?

Reply:
Hi, have you tried `Type[tf.keras.optimizers.Optimizer]`?

https://docs.python.org/3/library/typing.html#typing.Type

Reply:
Thanks @WindQAQ you're right!

Reply:
Thanks @WindQAQ. It may work 👍